android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java

```diff
@@ -260,6 +260,18 @@ private WebResourceResponse handleCapacitorHttpRequest(WebResourceRequest reques
             headers.put(header.getKey(), header.getValue());
         }
 
+        // a workaround for the following android web view issue:
+        // https://issues.chromium.org/issues/40450316
+        // x-cap-user-agent contains the user agent set in JavaScript
+        if (headers.has("x-cap-user-agent")) {
+            String userAgentValue = headers.getString("x-cap-user-agent");
+            if (userAgentValue != null && !userAgentValue.isEmpty()) {
+                headers.put("User-Agent", userAgentValue);
+            }
+
+            headers.remove("x-cap-user-agent");
+        }
```

Comment:
You can skip the `has` check since we still have to check for `getString` being null.
```suggestion
        String userAgentValue = headers.getString("x-cap-user-agent");
        if (userAgentValue != null && !userAgentValue.isEmpty()) {
            headers.put("User-Agent", userAgentValue);
            headers.remove("x-cap-user-agent");
        }
```